frontend/packages/client/src/hooks/useFileUploader.js

```diff
@@ -1,48 +1,47 @@
-import { useCallback, useReducer } from 'react';
 import { useErrorHandlerContext } from 'contexts/ErrorHandler';
 import { checkResponse } from 'utils';
-import { INITIAL_STATE, defaultReducer } from '../reducers';
+import { useMutation } from '@tanstack/react-query';
 
 export default function useFileUploader({ useModalNotifications = true } = {}) {
-  const [state, dispatch] = useReducer(defaultReducer, {
-    ...INITIAL_STATE,
-    loading: false,
-  });
   const { notifyError } = useErrorHandlerContext();
 
-  const uploadFile = useCallback(
+  const {
+    mutateAsync: uploadFile,
+    isLoading,
+    isError,
+    isSuccess,
+    data,
+    error,
+  } = useMutation(
     async (image) => {
-      dispatch({ type: 'PROCESSING' });
       const formData = new FormData();
```

Comment:
Can we move this to a function in our api directory?

Reply:
updated